protocol/x/perpetuals/types/perpetual.go

```diff
@@ -1,5 +1,29 @@
 package types
 
-func (m *Perpetual) GetId() uint32 {
-	return m.Params.Id
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	"github.com/pkg/errors"
+)
+
+func (p *Perpetual) GetId() uint32 {
+	return p.Params.Id
+}
+
+// Stateless validation on Perpetual params.
+func (p *PerpetualParams) Validate() error {
+	// Validate `ticker`.
+	if len(p.Ticker) == 0 {
+		return errors.WithStack(ErrTickerEmptyString)
+	}
+
+	// Validate `defaultFundingPpm`
+	defaultFundingPpm := lib.AbsInt32(p.DefaultFundingPpm)
+	if defaultFundingPpm > MaxDefaultFundingPpmAbs {
+		return sdkerrors.Wrap(
+			ErrDefaultFundingPpmMagnitudeExceedsMax,
+			lib.Int32ToString(p.DefaultFundingPpm))
```

Comment:
Did you mean to use `sdkerrors.Wrapf` with an error message, or was your intent to send back the raw value of `DefaultFundingPpm` as the error description?